crates/services/src/yield_stream.rs

```diff
@@ -0,0 +1,164 @@
+//! Stream that yields each `batch_size` items allowing other tasks to work.
+
+use futures::{
+    ready,
+    stream::Fuse,
+    Stream,
+    StreamExt,
+};
+use std::{
+    pin::Pin,
+    task::{
+        Context,
+        Poll,
+    },
+};
+
+pin_project_lite::pin_project! {
+    /// Stream that yields each `batch_size` items.
+    #[derive(Debug)]
+    #[must_use = "streams do nothing unless polled"]
+    pub struct YieldStream<St: Stream> {
+        #[pin]
+        stream: Fuse<St>,
```

Comment:
Is there a reason to fuse the stream here or are we just extra cautious? From what I can see, we guarantee that we won't call `poll_next` on this stream after it has returned `Poll::Ready(None)`, unless our `poll_next` is called after it itself has completed - which feels like it should be acceptable behavior, since we're just wrapping the behavior of the underlying stream.

Reply:
I tried to duplicate the logic of the `Chunk` iterator and, for stability, decided to continue return `None` if we did that in the past. In this case, the stream is predictable.

Reply:
Oh I see the chunk iterator now [(link](https://docs.rs/futures-util/0.3.31/src/futures_util/stream/stream/chunks.rs) for reference). Yeah it's very similar. I wonder why they decided to fuse the stream there 🤔